miqi/agent/tools/graph_render.py

```
@@ -0,0 +1,814 @@
+        f'x2="{layout["canvas_w"] - _PAD}" y2="{layout["canvas_h"] - _LEGEND_H}" '
+        f'stroke="#EEEEEE"/>'
+        f'<text x="{_PAD}" y="{layout["canvas_h"] - 14}" font-size="10.5" fill="#888">'
+        f"图例：{legend_items}</text>"
+        f"</svg>"
+    )
+    return svg
+
+
+# ── HTML 渲染（单文件交互版）─────────────────────────────────────────────
+def _html_escape_for_script(text: str) -> str:
+    """JS 字符串转义：确保 </script> 不会提前闭合。"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)